Bottom: dba117a1e565a95f377838325fb9f98f308c2dcf
Top:    b8a80c7b8d265cdd77ca0ecf0c172fe408b515ff
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-05 21:05:45 +0100

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 3994c9e..c490c36 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1120,7 +1120,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
